src/restconfig/src/main/java/org/geoserver/rest/security/AuthenticationProviderRestController.java

```diff
@@ -0,0 +1,425 @@
+    }
+
+    /// ///////////////////////////////////////////////////////////////////////
+    /// Implementation
+
+    private List<AuthProvider> listAuthProviders() {
+        try {
+            checkAuthorisation();
+            ArrayList<String> providerNames = new ArrayList<>(securityManager.listAuthenticationProviders());
+            return providerNames.stream()
+                    .map(this::loadProviderOrError)
+                    .peek(p -> {
+                        int index = providerNames.indexOf(p.getName());
+                        p.setPosition(index);
+                        p.setDisabled(index == -1);
+                    })
+                    .collect(toList());
+        } catch (IOException ex) {
+            throw new CannotReadConfiguration("All", ex);
+        }
+    }
+
+    private AuthProvider authProviderByName(String providerName) {
+        try {
+            checkArgument(!Strings.isNullOrEmpty(providerName), "Provider name cannot be null or empty");
+
+            SecurityAuthProviderConfig provider = securityManager.loadAuthenticationProviderConfig(providerName);
+            if (provider == null) {
+                throw new UnknownProvider(providerName);
+            }
+
+            AuthProvider authProvider = new AuthProvider(provider);
+            ArrayList<String> names = new ArrayList<>(securityManager.listAuthenticationProviders());
+            int index = names.indexOf(providerName);
+            authProvider.setPosition(index);
+            authProvider.setDisabled(index == -1);
+
+            return authProvider;
+        } catch (IOException e) {
+            throw new CannotReadConfiguration(providerName, e);
+        }
+    }
+
+    private void createAuthProvider(AuthProvider authProvider) {
+
+        try {
+            checkArgument(authProvider != null, "AuthProvider cannot be null");
+        } catch (IllegalArgumentException ex) {
+            throw new InvalidData(ex);
+        }
+
+        try {
+            if (authProvider.getPosition() == -1 && !authProvider.isDisabled()) {
+                throw new IllegalPosition(authProvider.getPosition(), authProvider.isDisabled());
+            }
+
+            if (Objects.nonNull(authProvider.getId())
+                    || Objects.nonNull(authProvider.getConfig().getId())) {
+                throw new IllegalArgumentException("Cannot create a new AuthProvider with an id");
+            }
+
+            if (Strings.isNullOrEmpty(authProvider.getName())) {
+                throw new IllegalArgumentException("Cannot create a new AuthProvider with without a name");
+            }
+
+            if (Strings.isNullOrEmpty(authProvider.getClassName())) {
+                throw new IllegalArgumentException("Cannot create a new AuthProvider with without a className");
+            }
+
+            // Ensure name
+            authProvider.getConfig().setName(authProvider.getName());
```

Comment:
I see a lot of authProvider.getConfig() around... why not allocate a local variable for it, when the code is using it heavily, and in the end, only intends to save it?
The AuthProvider wrapper is there just to carry the position (and eventually the enabled flag, assuming it has some use, see my other comments).